substrate/pkg/utils/discovery/tags.go

```diff
@@ -19,25 +19,24 @@ import (
 
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/service/ec2"
-	"github.com/awslabs/kubernetes-iteration-toolkit/substrate/pkg/apis/v1alpha1"
 )
 
 const (
 	OwnerTagKey = "kit.aws/substrate"
 )
 
-func Tags(substrate *v1alpha1.Substrate, name *string) []*ec2.Tag {
+func Tags(ownerName string, name *string) []*ec2.Tag {
 	return []*ec2.Tag{
-		{Key: aws.String(OwnerTagKey), Value: aws.String(substrate.Name)},
+		{Key: aws.String(OwnerTagKey), Value: aws.String(ownerName)},
 		{Key: aws.String("Name"), Value: name},
 	}
 }
 
-func Filters(substrate *v1alpha1.Substrate, optionalName ...*string) (filters []*ec2.Filter) {
+func Filters(name string, optionalName ...*string) (filters []*ec2.Filter) {
```

Comment:
Try not to change these functions but add new ones since these functions are being used in many files, we can avoid changing all the files. Changing all the files requires us to test all the functionality to make sure we are not breaking anything.

Suggestion add two new methods
```
func NameTag(ownerName, name *string) []*ec2.Tag {}
func FilterUsingName(name *string) (filters []*ec2.Filter) {}
```

Reply:
> Try not to change these functions but add new ones since these functions are being used in many files, we can avoid changing all the files. Changing all the files requires us to test all the functionality to make sure we are not breaking anything.
> 
> Suggestion add two new methods
> 
> ```
> func NameTag(ownerName, name *string) []*ec2.Tag {}
> func FilterUsingName(name *string) (filters []*ec2.Filter) {}
> ```

The scope of the usage of `Tags` & `Filters` is only 2 more files larger than our change. (`infrastructure/natgateway.go` & cluster/address.go), I think it's small enough change to change the `Tags` & `Filters` function.